src/shared/getNpmUserInfo.test.ts

```diff
@@ -0,0 +1,57 @@
+import { describe, expect, it, vi } from "vitest";
+
+import {
+	ErrorResult,
+	SuccessResult,
+	getNpmUserInfo,
+} from "./getNpmUserInfo.js";
+
+const mock$ = vi.fn();
+
+vi.mock("execa", () => ({
+	get $() {
+		return mock$;
+	},
+}));
+
+const mockNpmUser = vi.fn();
+
+vi.mock("npm-user", () => ({
+	get default() {
+		return mockNpmUser;
+	},
+}));
+
+const npmUsername = "test-npm-username";
+
+describe("getNpmUserInfo", () => {
+	it("returns an error result when npm whoami fails", async () => {
+		mock$.mockRejectedValue({ stderr: "Oh no!" });
+		const result = (await getNpmUserInfo()) as ErrorResult;
+
+		expect(result.succeeded).toBe(false);
+		expect(result.reason).toBe(
+			"Could not populate npm user. Failed to run npm whoami."
+		);
```

Comment:
[Nitpick] You can combine these assertions with a `.toEqual` to avoid needing the `as` assertion:

```suggestion
		const result = await getNpmUserInfo();

		expect(result).toEqual({
			reason: "Could not populate npm user. Failed to run npm whoami.",
			succeeded: false,
		});
```

(here and elsewhere)